cpp/oneapi/dal/algo/pca/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -39,25 +39,65 @@ static result_t infer(const context_gpu& ctx, const descriptor_t& desc, const in
     const auto data = input.get_data();
     auto model = input.get_model();
     auto eigenvectors = model.get_eigenvectors();
+    auto means = model.get_means();
+    auto eigenvalues = model.get_eigenvalues();
+
     const std::int64_t row_count = data.get_row_count();
+    const std::int64_t col_count = data.get_column_count();
     const std::int64_t component_count = get_component_count(desc, data);
     dal::detail::check_mul_overflow(row_count, component_count);
 
     const auto data_nd = pr::table2ndarray<Float>(queue, data, sycl::usm::alloc::device);
+    auto means_nd = pr::table2ndarray_1d<Float>(queue, means, sycl::usm::alloc::device);
+    auto mean_centered_data_nd =
+        pr::ndarray<Float, 2>::empty(queue, { row_count, col_count }, sycl::usm::alloc::device);
     const auto eigenvectors_nd =
         pr::table2ndarray<Float>(queue, eigenvectors, sycl::usm::alloc::device);
 
+    pr::elementwise_difference(queue, row_count, data_nd, means_nd, mean_centered_data_nd);
```

Comment:
I would suggest to use 
auto 'event_name' = pr::elementwise_difference(queue, row_count, data_nd, means_nd, mean_centered_data_nd);
and provide this event in gemm as deps